packages/EasyErrorHandler/src/Bridge/Symfony/Builder/ApiPlatformValidationErrorResponseBuilder.php

```diff
@@ -32,11 +34,19 @@ public static function supports(Throwable $throwable): bool
     {
         $message = $throwable->getMessage();
 
+        // TODO: refactor in 5.0. Use the ApiPlatform\Symfony\Bundle\ApiPlatformBundle class only.
+        $invalidArgumentExceptionClass = stdClass::class;
+        if (\class_exists(InvalidArgumentException::class)) {
+            $invalidArgumentExceptionClass = InvalidArgumentException::class;
+        }
+
         return match ($throwable::class) {
-            InvalidArgumentException::class => \preg_match(self::MESSAGE_PATTERN_TYPE_ERROR, $message) === 1,
+            $invalidArgumentExceptionClass, LegacyInvalidArgumentException::class =>
+                \preg_match(self::MESSAGE_PATTERN_TYPE_ERROR, $message) === 1,
             MissingConstructorArgumentsException::class =>
                 \preg_match(self::MESSAGE_PATTERN_NO_PARAMETER, $message) === 1,
             UnexpectedValueException::class =>
+                \preg_match(self::MESSAGE_PATTERN_TYPE_ERROR, $message) === 1 ||
```

Comment:
What is the reason to add it in that case?

Reply:
I didn't investigate deeply. It looks like the API platform started returning the Symfony Serializer exception during deserialization.

Reply:
We should exactly understand what we do. So please investigate why this error is possible here and why we didn't add it initially.

Reply:
Before 2.7 API Platform throw `InvalidArgumentException` in `\ApiPlatform\Core\Serializer\AbstractItemNormalizer::validateType`

Now it throws `UnexpectedValueException` in `\ApiPlatform\Serializer\AbstractItemNormalizer::validateType`

Reply:
Well, does it then mean we don't need to check `$invalidArgumentExceptionClass` above and we need only `LegacyInvalidArgumentException` there?

Reply:
We need both, in version `2.7` `InvalidArgumentException` at least uses in `\ApiPlatform\Serializer\AbstractItemNormalizer::denormalizeCollection`.